.github/workflows/ci-post-land.yml

```diff
@@ -0,0 +1,351 @@
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ENV_DIEM_S3_AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.ENV_DIEM_S3_AWS_REGION }}
+
+  build_ci_base_docker_image:
+    runs-on: ubuntu-latest-xl
+    if: ${{ needs.prepare.outputs.base-image-changes == 'true' && github.event_name == 'push' }}
+    continue-on-error: false
+    env:
+      DOCKERHUB_ORG: diem
+    environment:
+      name: Docker
+      url: https://hub.docker.com/u/diem
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 #get all the history!!!
+      - name: Git Hooks and Checks
+        run: ./scripts/git-checks.sh
+      - name: build image
+        run: docker build -f docker/ci/github/Dockerfile -t ${{ env.DOCKERHUB_ORG }}/build_environment:${{ needs.prepare.outputs.changes-target-branch }} .
+      - name: Sign in to dockerhub, install image signing cert.
+        uses: ./.github/actions/dockerhub_login
+        with:
+          username: ${{ secrets.ENV_DOCKERHUB_USERNAME }}
+          password: ${{ secrets.ENV_DOCKERHUB_PASSWORD }}
+          key_material: ${{ secrets.ENV_DOCKERHUB_KEY_MATERIAL }}
+          key_name: ${{ secrets.ENV_DOCKERHUB_KEY_NAME }}
+          key_password: ${{ secrets.ENV_DOCKERHUB_KEY_PASSWORD }}
+      - name: Push to dockerhub.
+        run: |
+          if [[ "$DOCKERHUB_LOGGED_IN" == true ]]; then
+            disable_content_trust=true
+            if [[ "$DOCKERHUB_CAN_SIGN" == true ]]; then
+              disable_content_trust=false
+            fi
+            docker push --disable-content-trust=${disable_content_trust} ${{ env.DOCKERHUB_ORG }}/build_environment:${{ needs.prepare.outputs.changes-target-branch }}
+          fi
+        env:
+          DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ${{ secrets.ENV_DOCKERHUB_KEY_PASSWORD }}
+
+  run-cluster-test-pre-release-suite:
+    name: Run the pre-release suite of Cluster Test
+    runs-on: self-hosted
+    if: ${{  startsWith(needs.prepare.outputs.changes-target-branch, 'release') }}
+    # The pre-release suite run time varies 1~1.5 hr.
+    timeout-minutes: 120
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0 #get all the history!!!
+      - name: set_env
+        id: set_env
+        run: |
+          HEAD_GIT_REV=$(git rev-parse --short=8 HEAD)
+          echo "HEAD_GIT_REV=$HEAD_GIT_REV" >> $GITHUB_ENV
+          IMAGE_TAG=$(echo ${GITHUB_REF#refs/heads/})_$HEAD_GIT_REV
+          echo $IMAGE_TAG
+          echo "IMAGE_TAG=$IMAGE_TAG" >> $GITHUB_ENV
+      - name: poll_images
+        # Poll until images are ready
+        env:
+          AWS_REGION: us-west-2
+          RETRIES: 30
+        run: |
+          set +e
```

Comment:
fwiw an embedded script (rather than a name) feels a bit rough here.

Reply:
Just a port of the job.   But yeah, there's a lot of debt here.  Since folks have been working to replace cluster test I'm not trying to fix any of this and take on all the cluster test debt.